internal/provider/networks_wireless_ssids_resource.go

```diff
@@ -0,0 +1,1727 @@
+	}
+	request.SetRadiusServers(radiusServers)
+
+	radiusAccountingServers := []openApiClient.UpdateNetworkWirelessSsidRequestRadiusAccountingServersInner{}
+	for _, radius := range ssid.RadiusAccountingServers {
+		radiusServer := openApiClient.NewUpdateNetworkWirelessSsidRequestRadiusAccountingServersInner(radius.Host.ValueString())
+		radiusServer.SetPort(int32(radius.Port.ValueInt64()))
+		radiusServer.SetSecret(radius.Secret.ValueString())
+		radiusServer.SetCaCertificate(radius.CaCertificate.ValueString())
+		radiusServer.SetRadsecEnabled(radius.RadsecEnabled.ValueBool())
+		radiusAccountingServers = append(radiusAccountingServers, *radiusServer)
+	}
+	request.SetRadiusAccountingServers(radiusAccountingServers)
+
+	apTagsAndVlanIds := []openApiClient.UpdateNetworkWirelessSsidRequestApTagsAndVlanIdsInner{}
+	for _, vlanID := range ssid.ApTagsAndVlanIds {
+		apTagsAndVlanId := openApiClient.NewUpdateNetworkWirelessSsidRequestApTagsAndVlanIdsInner()
+		apTagsAndVlanId.SetVlanId(int32(vlanID.VlanId.ValueInt64()))
+		tags := []string{}
+		for _, tag := range vlanID.Tags.Elements() {
+			tags = append(tags, tag.String())
+		}
+		apTagsAndVlanId.SetTags(tags)
+		apTagsAndVlanIds = append(apTagsAndVlanIds, *apTagsAndVlanId)
+	}
+	request.SetApTagsAndVlanIds(apTagsAndVlanIds)
+
+	_, httpResp, err := r.client.WirelessApi.UpdateNetworkWirelessSsid(context.Background(), data.NetworkID.ValueString(), data.Number.ValueString()).UpdateNetworkWirelessSsidRequest(*request).Execute()
+	// Check for API success response code
+	if httpResp.StatusCode == 404 {
+		resp.Diagnostics.AddWarning(
+			"no ssid number or network found in API",
+			tools.HttpDiagnostics(httpResp),
+		)
+
+	} else if httpResp.StatusCode != 200 {
+		resp.Diagnostics.AddError(
+			"Unexpected HTTP Response Status Code",
+			fmt.Sprintf("%v", httpResp.StatusCode),
+		)
+
+		// HTTP 400 counts as an error so moving this here
+		// If there was an error during API call, add it to diagnostics.
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"HTTP Client Failure",
+				tools.HttpDiagnostics(httpResp),
+			)
+			return
+		}
+
+		// If there were any errors up to this point, log the state data and return.
+		if resp.Diagnostics.HasError() {
+			resp.Diagnostics.AddError("State Data", fmt.Sprintf("\n%v", data))
+			return
+		}
+
+		// Decode the HTTP response body into your data model.
+		// If there's an error, add it to diagnostics.
+		if err = json.NewDecoder(httpResp.Body).Decode(&data.SSID); err != nil {
+			resp.Diagnostics.AddError(
+				"JSON decoding error",
+				fmt.Sprintf("%v\n", err.Error()),
+			)
+			return
+		}
+	}
+
+	data.Id = jsontypes.StringValue("example-id")
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data.SSID)...)
+
+	// Write logs using the tflog package
+	tflog.Trace(ctx, "updated resource")
+}
+
+func (r *NetworksWirelessSsidsResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
```

Comment:
Regarding the delete function for resources which are leveraging a POST/PUT API call we do not need to read the config from Data, nor do we have to marshal the HTTP response into the internal data model.

What needs to be done is this case is that we reset the Device/Resource to a default state, so we need only to create a new payload with null values wherever possible to erase the old configuration. Once this is done successfully we can simply remove the resource from state with this:


```
resp.State.RemoveResource(ctx)
```